File: java/src/test/java/problems301_400/DecodeStringTest.java
package problems301_400;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class DecodeStringTest extends SampledTest {
    @Test
    public void test() {
        for (DecodeString solution : DecodeString.values()) {
            for (DecodeStringSample sample : getSamples(DecodeStringSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.decodeString(sample.input.s))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems301_400/DecodeStringTest.kt
package problems301_400

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class DecodeStringTest : SampledTest() {
    @Test
    fun test() =
        DecodeString.entries.forEach { solution ->
            getSamples<Array<DecodeStringSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.decodeString(input.s))
                    .isEqualTo(output)
            }
        }
}
